src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,64 +113,67 @@ public NuGetProject DefaultNuGetProject
                 throw new ArgumentNullException(nameof(projectSystemFactory));
             }
 
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _serviceProvider = serviceProvider;
             _projectSystemCache = projectSystemCache;
             _projectSystemFactory = projectSystemFactory;
+            _logger = logger;
+        }
 
-            _dte = ServiceLocator.GetInstance<DTE>();
-            _vsSolution = ServiceLocator.GetGlobalService<SVsSolution, IVsSolution>();
-            _vsMonitorSelection = ServiceLocator.GetGlobalService<SVsShellMonitorSelection, IVsMonitorSelection>();
+        public async Task InitializeAsync(IAsyncServiceProvider site)
+        {
```

Comment:
Due to `high risk of a deadlock in VS` any threading work in MEF is forbidden. That's not an option.
Then if you take closer look at `InitializeAsync` it only attaches to events which can be safely delayed. `VsSolutionManager` is fully functional right after being instantiated, IE even before `InitializeAsync` is called.